packages/gasket-plugin-webpack/index.js

```diff
@@ -6,29 +7,36 @@ const { name, devDependencies } = require('./package');
 /**
 * Creates the webpack config
 * @param  {Gasket} gasket The Gasket API
-* @param {Object} webpackConfig Initial webpack config
-* @param {Object} data Additional info
+* @param {Object} initConfig Initial webpack config
+* @param {Object} nextOptions Additional info
 * @returns {Object} Final webpack config
 */
-function initWebpack(gasket, webpackConfig, data) {
-  const { execSync, config } = gasket;
+function initWebpack(gasket, initConfig, nextOptions) {
+  const { execSync, execWaterfallSync, config } = gasket;
+
+  const standardPlugins = { plugins: [new WebpackMetricsPlugin({ gasket })] };
 
   const chain = new WebpackChain();
-  execSync('webpackChain', chain, data);
+  execSync('webpackChain', chain, nextOptions);
 
-  //
-  // Merge defaults with gasket.config webpack.
-  //
-  webpackConfig = webpackMerge.smart(
-    webpackConfig,
-    { plugins: [new WebpackMetricsPlugin({ gasket })] },
-    chain.toConfig(),     // Webpack chain from plugins (partial)
-    config.webpack || {}  // Webpack config from user (partial)
+  const baseConfig = webpackMerge.smart(
+    initConfig,
+    standardPlugins,
+    chain.toConfig(),         // From webpackChain (partial)
+    config.webpack || {}      // From gasket config file (partial)
   );
 
-  const configs = execSync('webpack', webpackConfig, data).filter(Boolean);
+  const configPartials = execSync('webpack', baseConfig, nextOptions).filter(Boolean);
+  const mergedConfig = webpackMerge.smart(
+    baseConfig,
+    ...configPartials
+  );
 
-  return webpackMerge.smart(webpackConfig, ...configs);
+  return execWaterfallSync(
+    'webpackConfig',
+    mergedConfig,
+    { webpack, nextOptions, webpackMerge }
```

Comment:
This should next be Next-specific. See the previous comment. If we call this the context, instead just add the helpers to it:
```suggestion
    { ...context, webpack, webpackMerge }
```

Reply:
Updated